### PR TITLE
drivers/oracle: standardize tests on testh harness; trim driver READMEs

### PR DESCRIPTION
**Pull request by neilotoole**
## Summary

Move the Oracle driver's integration tests onto the standard `testh` / `@sakila_or` harness that every other SQL driver already uses, and remove the bespoke test scaffolding that had accumulated when the driver was added. Also trims the Oracle and ClickHouse maintainer READMEs down to driver-specific content that isn't already covered elsewhere.

## Why

Oracle was the only driver reading DB connection details from `os.Getenv` in tests. It had invented `SQ_TEST_ORACLE_DSN` (plus `SQ_TEST_POSTGRES_DSN` for its cross-driver copy test), each with hardcoded `localhost` defaults and a parallel set of skip/source helpers — while *also* wiring up the standard `@sakila_or` handle. Two of its four integration files already used the standard harness; this brings the other two in line so there's a single, consistent mechanism.

## Changes

**Tests**
- Remove `SQ_TEST_ORACLE_DSN`, `SQ_TEST_POSTGRES_DSN`, their hardcoded defaults, and the `skipIfNoOracle` / `getTestSource` / `skipIfNoPostgres` / `openOracleForInternalTest` helpers.
- `oracle_test.go` uses `testh.NewWith(t, sakila.Ora)` (and `sakila.Pg` for the cross-driver copy), matching the existing idiom in `metadata_test.go`, `oracle_sqldriver_test.go`, and `TestDBSemver`. Integration tests now skip cleanly when `SQ_TEST_SRC__SAKILA_OR` is unset.
- `TestGetSourceMetadata_NoSchema` (which needs a live DB but exercises an unexported branch) moves to the external test package via a small `export_test.go` seam — `testh` imports this driver, so an internal test can't import `testh` without an import cycle.
- `TestMetadataHelpers_ErrorPaths` becomes a deterministic unit test: closed `*sql.DB`, no env var, always runs.

**Docs**
- Trim `drivers/oracle/README.md` and `drivers/clickhouse/README.md` to maintainer content not already on the sq.io driver pages / `docs/DRIVERS.md` / `go.mod`.
- Fix two stale items in the ClickHouse README (a hardcoded dependency version and a link to a gitignored memo).

## Verification

Run against a live `sakiladb/oracle` instance (Postgres also available, so the cross-driver copy test executed rather than skipped):

- Full `./drivers/oracle/...` package: **67/67 pass**.
- `go vet ./drivers/oracle/...`: clean.
- `golangci-lint run ./drivers/oracle/...`: 0 issues.
- Confirmed clean skip when `SQ_TEST_SRC__SAKILA_OR` is unset.

No CI changes needed — no CI job referenced the removed env vars.

### CODE REVIEW

File: drivers/oracle/export_test.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package oracle
+
+// Symbols exported for use by the external oracle_test package.
+
+// GetSourceMetadata exposes getSourceMetadata so that external tests can cover
+// the noSchema=true branch, which the exported grip.SourceMetadata path does
+// not reach.
+var GetSourceMetadata = getSourceMetadata
```
[the remark this answers is not in the document]
Comment (neilotoole) [reply drivers/oracle/export_test.go:15]: Keeping the `var` alias here: it is the idiomatic Go test-export pattern (the standard library's own `export_test.go` files do `var ExportFoo = foo`), and reassignability is not a concern for a symbol that only exists in test builds. A wrapper func would duplicate the four-arg signature and pull in four imports for no behavioral gain.

File: drivers/clickhouse/README.md [hunk 2]
```diff
@@ -133,246 +90,49 @@ which maps to `kind.Text`.
...
+- **Rows affected**: ClickHouse reports `0` for all DML. sq surfaces this as
+  `dialect.RowsAffectedUnsupported` (-1); text output shows
+  "rows affected: unsupported".
+
+### Mutations are asynchronous by default
+
+ClickHouse has no standard `UPDATE`/`DELETE`; it uses "lightweight mutations"
+via `ALTER TABLE ... UPDATE/DELETE`, which are **asynchronous by default** —
+the statement returns before rows change, so a subsequent `SELECT` may see
+stale data.
```
[the remark this answers is not in the document]
Comment (neilotoole) [reply drivers/clickhouse/README.md:122]: Fixed in 4ffe0ae9 (em dashes replaced with a colon/comma).

File: drivers/clickhouse/README.md [hunk 3]
```diff
@@ -388,572 +148,33 @@ export SQ_TEST_SRC__SAKILA_CH='clickhouse://sakila:p_ssW0rd@localhost:9000/sakil
...
-`StmtExecer.Close()` was added in
-`libsq/driver/record_test.go:TestStmtExecer_Close_NilStmt`.
+<!-- markdownlint-enable MD013 -->
 
-**Status**: Resolved.
-
-### Batch Insert Edge Case Tests (2026-02-15)
+For #1/#2, `TestDriver_CreateTable_Minimal` and `TestOutputRaw` are skipped for
+ClickHouse (the `kind.Bytes` data is preserved on disk — `String` is
+binary-safe — but the Go type and sq kind change on readback).
```
[the remark this answers is not in the document]
Comment (neilotoole) [reply drivers/clickhouse/README.md]: Fixed in 4ffe0ae9 (em dashes replaced with a colon/comma).
